src/memory/client/MemoryBrowser.tsx

```diff
@@ -77,21 +77,52 @@ interface State {
   bytesPerRow: number;
   bytesPerGroup: number;
   endianness: 'le' | 'be';
+  childNames: { id: string; name: string }[];
 }
 
 export class MemoryBrowser extends React.Component<Props, State> {
   private addressReq = '';
   private lengthReq = '512';
+  private childReq = '0';
+  public static firtTime: boolean = true;
 
   constructor(props: Props) {
     super(props);
     this.state = {
       bytesPerRow: 32,
       bytesPerGroup: 8,
       endianness: 'le',
+      childNames: [],
     };
   }
 
+  async getAvailableChilds() {
+    if (this.state.childNames.length === 0) {
+      try {
+        const result = await messageBroker.send({
+          command: 'ReadMemory',
+          args: {
+            address: this.addressReq,
+            length: parseInt(this.lengthReq),
+            child: this.childReq,
+          },
+        });
+        let childArray: {
+          id: any;
+          name: any;
+        }[] = [];
+        result.result?.child?.map(function (val: any, index: any) {
```

Comment:
This isn't how to use map, map should transform one thing to another, e.g.:

```typescript
const childrenArray = result.result?.child?.map((val, index) => {id: index, name: val});
```

Please see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map for a comprehensive overview of map in javascript.

Reply:
The input of this function is an array and the output is an array of objects for example:
input: [RX, RL, RZ]
output: [{id: 0, name: RX}, {id: 1, name: RL}, {id: 2, name: RZ}]

Reply:
Yes, I understand that, but map is normally used to map from one type to another. I gave you the standard way of writing that code, but you didn't change it, not explain what your issue is.